packages/eas-cli/src/submit/ArchiveSource.ts

```diff
@@ -177,9 +181,9 @@ async function handlePathSourceAsync(
   }
 
   Log.log('Uploading your app archive to the Expo Submission Service');
-  const uploadUrl = await uploadAppArchiveAsync(graphqlClient, source.path);
+  const bucketKey = await uploadAppArchiveAsync(graphqlClient, source.path);
   return {
-    url: uploadUrl,
+    resolvedArchiveSource: { type: SubmissionArchiveSourceType.GcsSubmitArchive, bucketKey },
     source,
```

Comment:
And:
```suggestion
    source: { type: SubmissionArchiveSourceType.GcsSubmitArchive, bucketKey },
```

If I understand correctly, this needs another `Archive...` interface. 
I don't think we should have two almost identical fields in the return object.